File: app/src/main/java/com/cyl/musiclake/data/db/Playlist.java
//package com.cyl.musiclake.data.db;
//
//import com.cyl.musiclake.data.db.Music;
//
//import org.litepal.annotation.Column;
//import org.litepal.crud.DataSupport;
//
//import java.io.Serializable;
//import java.util.ArrayList;
//import java.util.List;
//
///**
// * 功能：本地歌单
// * 作者：yonglong on 2016/9/13 21:59
// * 邮箱：643872807@qq.com
// * 版本：2.5
// */
//public class Playlist extends DataSupport implements Serializable {
//    //歌单id
//    private String pid;
//    //歌单名
//    private String name;
//    //歌单名
//    @Column(defaultValue = "0")
//    private int count;
//    //更新日期
//    private long updateDate;
//    //创建日期
//    private long date;
//    //描述
//    private String des;
//    @Column(defaultValue = "date asc")
//    private String order;
//    private String coverUrl;
//    //类型 0：本地歌单 1：在线歌单
//    private int type;
//
//    //歌曲集合
//    private List<Music> musicList = new ArrayList<>();
//
//    public Playlist() {
//    }
//
//    public Playlist(String id, String name) {
//        this.pid = id;
//        this.name = name;
//    }
//
//    public Playlist(String id, String name, int count, long date, String order) {
//        this.pid = id;
//        this.name = name;
//        this.count = count;
//        this.date = date;
//        this.order = order;
//    }
//
//    public long getUpdateDate() {
//        return updateDate;
//    }
//
//    public void setUpdateDate(long updateDate) {
//        this.updateDate = updateDate;
//    }
//
//    public int getType() {
//        return type;
//    }
//
//    public void setType(int type) {
//        this.type = type;
//    }
//
//    public String getId() {
//        return pid;
//    }
//
//    public void setId(String id) {
//        this.pid = id;
//    }
//
//    public String getName() {
//        return name;
//    }
//
//    public void setName(String name) {
//        this.name = name;
//    }
//
//    public int getCount() {
//        return count;
//    }
//
//    public void setCount(int count) {
//        this.count = count;
//    }
//
//    public long getDate() {
//        return date;
//    }
//
//    public void setDate(long date) {
//        this.date = date;
//    }
//
//    public String getDes() {
//        return des;
//    }
//
//    public void setDes(String des) {
//        this.des = des;
//    }
//
//    public String getOrder() {
//        return order;
//    }
//
//    public void setOrder(String order) {
//        this.order = order;
//    }
//
//    public String getCoverUrl() {
//        return coverUrl;
//    }
//
//    public void setCoverUrl(String coverUrl) {
//        this.coverUrl = coverUrl;
//    }
//
//    public List<Music> getMusicList() {
//        return musicList;
//    }
//
//    public void setMusicList(List<Music> musicList) {
//        this.musicList = musicList;
//    }
//
//    @Override
//    public String toString() {
//        return name;
//    }
//}

File: app/src/main/java/com/cyl/musiclake/data/db/Playlist.kt
package com.cyl.musiclake.data.db

import org.litepal.crud.LitePalSupport
import java.io.Serializable
import java.util.*

/**
 * 功能：本地歌单
 * 作者：yonglong on 2016/9/13 21:59
 * 邮箱：643872807@qq.com
 * 版本：2.5
 */
class Playlist : LitePalSupport, Serializable {
    var id: Long = 0
    //歌单id
    var pid: String? = null
    //歌单名
    var name: String? = null
    //歌单名
    var count: Int = 0
    //更新日期
    var updateDate: Long = 0
    //创建日期
    var date: Long = 0
    //描述
    var des: String? = null
    //排列顺序
    var order: String? = null
    //封面
    var coverUrl: String? = null
    //类型 0：本地歌单 1：在线歌单
    var type: Int = 0

    //歌曲集合
    var musicList: List<Music> = ArrayList()

    constructor() {}

    constructor(pid: String?, name: String?) {
        this.pid = pid
        this.name = name
    }

    constructor(pid: String?, name: String?, order: String ="updateDate desc") {
        this.pid = pid
        this.name = name
    }

    override fun toString(): String {
        return "Playlist(id=$id, name=$name, count=$count, updateDate=$updateDate, date=$date, des=$des, order=$order, coverUrl=$coverUrl, type=$type, musicList=$musicList)"
    }

}

class MusicToPlaylist : LitePalSupport() {
    var id: Long = 0
    var pid: String? = null
    var mid: String? = null
    var count: Long = 0
    var updateDate: Long = 0
    var createDate: Long = 0
}